core/src/main/java/org/kiwix/kiwixmobile/core/data/local/KiwixRoomDatabase.kt

```diff
@@ -0,0 +1,60 @@
+/*
+ * Kiwix Android
+ * Copyright (c) 2022 Kiwix <android.kiwix.org>
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package org.kiwix.kiwixmobile.core.data.local
+
+import android.content.Context
+import androidx.multidex.BuildConfig
+import androidx.room.Database
+import androidx.room.Room
+import androidx.room.RoomDatabase
+import io.objectbox.BoxStore
+import io.objectbox.kotlin.boxFor
+import org.kiwix.kiwixmobile.core.dao.RecentSearchRoomDao
+import org.kiwix.kiwixmobile.core.dao.entities.RecentSearchRoomEntity
+
+@Suppress("UnnecessaryAbstractClass")
+@Database(entities = [RecentSearchRoomEntity::class], version = 1)
+abstract class KiwixRoomDatabase : RoomDatabase() {
+  abstract fun recentSearchRoomDao(): RecentSearchRoomDao
+
+  companion object {
+    private var db: KiwixRoomDatabase? = null
+    fun getInstance(context: Context, boxStore: BoxStore): KiwixRoomDatabase {
+      return db ?: synchronized(KiwixRoomDatabase::class) {
+        return@getInstance db
+          ?: Room.databaseBuilder(context, KiwixRoomDatabase::class.java, "KiwixRoom.db")
+            // We have already database name called kiwix.db in order to avoid complexity we named as
+            // kiwixRoom.db
+            .build().also {
+              if (!BuildConfig.BUILD_TYPE.contentEquals("fdroid")) {
```

Comment:
@MohitMaliFtechiz We aren't migrating the data in fdroid flavour.

Reply:
So you meant we are totally remove the object box dependency from fdroid build?

Reply:
Yes, we have too, otherwise we won't be able anymore to publish on FDroid. We need to be able to compile:
* With ObjectBox for PlayStore (during the time we want to allow user to migrate their data to Room)
* Without ObjectBox in all other scenarios.

We need therefore a special compilation target for Google in the Gradle

Reply:
* But we are initialise fdroid from ```baseRelease``` and baseRelease from ```debug``` .
* For debug we are adding object box dependency in core module.

`
    getByName("debug") {
      dependencies {
        implementation(Libs.objectbox_kotlin)
        implementation(Libs.objectbox_rxjava)
      }
    }
`

so it will automatically added in fdroid module as well.

Reply:
@MohitMaliFtechiz Done https://github.com/kiwix/kiwix-android/pull/3143/commits/62f8378a345885b4dc680adb13a33f381f0c0e47

Reply:
@MohitMaliFtechiz I think it's a regression. See https://github.com/kiwix/kiwix-android/issues/3282. After we fix it we can achieve it.

Reply:
@gouri-panda , i think this is import error like it's ```androidx.multidex.BuildConfig``` not ```org.kiwix.kiwixmobile.BuildConfig``` if we are using the ```org.kiwix.kiwixmobile.BuildConfig``` then we can get every build variants like in below image or if we want to get build type from ```androidx.multidex.BuildConfig```, so i think first we needs to set build type on ```androidx.multidex.BuildConfig```.

![Screenshot from 2023-03-31 17-48-45](https://user-images.githubusercontent.com/34593983/229118065-2c361787-b7be-4dc1-a088-c45bfaf55cfc.png)

Reply:
Hi @gouri-panda , I think you forgot to change the import here.

Reply:
I have did changes.